File: ports/perl/tests/render.pl
use strict;
use warnings;
use FindBin;
use lib "$FindBin::Bin/../lib";
use Hqtui;
while(my $line=<STDIN>) {
    my $c=JSON::PP->new->utf8->decode($line);
    my $s=Hqtui::Scene->new(width=>$c->{width},height=>$c->{height},theme=>$c->{theme});
    if($c->{screen}) {print $s->demo_frame($c->{screen},'hashes'),"\n";}
    else {$s->set($c->{tree});print $s->render('hashes'),"\n";}
    $s->close;
}

File: ports/ruby/tests/render.rb
require_relative '../lib/hqtui'
STDIN.each_line do |line|
  c = JSON.parse(line)
  s = Hqtui::Scene.new(width: c['width'], height: c['height'], theme: c['theme'])
  begin
    s.collapse(true) if c['collapsed']
    if c['screen']
      puts s.demo_frame(c['screen'], 'hashes')
    else
      s.set(c['tree'])
      puts s.render('hashes')
    end
  ensure
    s.close
  end
end
